### PR TITLE
Bump SDKs, add communication-common beta, update pnpm

### PR DESCRIPTION
**Pull request by JamesBurnside**
NOTE: this PR needs these two changes first:
- #3357
- #3356 

# What

- Bump `@azure/communication-calling` to 1.15.1-beta.1
- Bump `@azure/communication-chat` to 1.3.2-beta.3
- Introduce a beta flavor for `@azure/communication-common`
  - Put new MicrosoftBotIdentifier changes behind a new beta flag (`communication-common-beta-v3`)
  - Update checks to cope with the fact a participant ID could now be a MicrosoftBotIdentifier
  - Update `force-build-flavors` to expect the new `||` flag

# Why

Housekeeping

# How Tested

n/a - CI should test this

### CODE REVIEW

**Review by AmyL219 — APPROVED** [review]

File: packages/calling-stateful-client/src/CallAgentDeclarativeCommon.ts [hunk 1]
```diff
@@ -180,21 +180,27 @@ export abstract class ProxyCallAgentCommon {
       }
       case 'on': {
         return (...args: Parameters<AgentType['on']>): void => {
-          const isCallsUpdated = args[0] === 'callsUpdated';
+          // typescript is not smart enough to handle multiple overloads and pull the correct type here so force casting args
+          const event = args[0] as 'incomingCall' | 'callsUpdated' | 'connectionStateChanged';
```
[the remark this answers is not in the document]
Comment (JamesBurnside) [reply packages/calling-stateful-client/src/CallAgentDeclarativeCommon.ts:184]: Oh good point, let me move the listener type cast to inside the `if (isCallsUpdated) {`

[the remark this answers is not in the document]
Comment (JamesBurnside) [reply packages/calling-stateful-client/src/CallAgentDeclarativeCommon.ts:184]: moved

File: packages/communication-react/review/beta/communication-react.api.md [hunk 2]
```diff
@@ -3317,7 +3318,7 @@ export interface RecordingCallFeature {
 export interface RemoteParticipantState {
     callEndReason?: CallEndReason;
     displayName?: string;
-    identifier: CommunicationUserKind | PhoneNumberKind | MicrosoftTeamsUserKind | UnknownIdentifierKind;
+    identifier: CommunicationUserKind | PhoneNumberKind | MicrosoftTeamsUserKind | UnknownIdentifierKind | /* @conditional-compile-remove(communication-common-beta-v3) */ MicrosoftBotIdentifier;
```
Comment (AmyL219) [inline packages/communication-react/review/beta/communication-react.api.md:3321]: breaking change in API.md

File: packages/calling-stateful-client/review/beta/calling-stateful-client.api.md [hunk 3]
```diff
@@ -267,7 +268,7 @@ export interface RecordingCallFeature {
 export interface RemoteParticipantState {
     callEndReason?: CallEndReason;
     displayName?: string;
-    identifier: CommunicationUserKind | PhoneNumberKind | MicrosoftTeamsUserKind | UnknownIdentifierKind;
+    identifier: CommunicationUserKind | PhoneNumberKind | MicrosoftTeamsUserKind | UnknownIdentifierKind | /* @conditional-compile-remove(communication-common-beta-v3) */ MicrosoftBotIdentifier;
```
[the remark this answers is not in the document]
Comment (JoshuaLai) [reply packages/calling-stateful-client/review/beta/calling-stateful-client.api.md:271]: In this case it should be fine. This is a new type exposed by the common sdk. So exposing it in the stateful client makes sense.

[the remark this answers is not in the document]
Comment (JoshuaLai) [reply packages/calling-stateful-client/review/beta/calling-stateful-client.api.md:271]: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/communication/communication-common/CHANGELOG.md#300-beta1-2023-03-27 Changelog here

[the remark this answers is not in the document]
Comment (JoshuaLai) [reply packages/calling-stateful-client/review/beta/calling-stateful-client.api.md:271]: Actually taking a second pass at this I think this is exposing the wrong type here: 
```suggestion
    identifier: CommunicationUserKind | PhoneNumberKind | MicrosoftTeamsUserKind | UnknownIdentifierKind | /* @conditional-compile-remove(communication-common-beta-v3) */ MicrosoftBotIdentifierKind;
```

[the remark this answers is not in the document]
Comment (JoshuaLai) [reply packages/calling-stateful-client/review/beta/calling-stateful-client.api.md:271]: Source: https://github.com/Azure/azure-sdk-for-js/blob/7528a7a1117421e1595ac006557a967ae1a2dee6/sdk/communication/communication-common/src/identifierModels.ts#L196

[the remark this answers is not in the document]
Comment (AmyL219) [reply packages/calling-stateful-client/review/beta/calling-stateful-client.api.md]: Good catch. Updated

**Review by dmceachernmsft — APPROVED** [review]

**Review by JoshuaLai — APPROVED** [review]

**Pull request merged**